### PR TITLE
add snappy and zstd compression support

### PR DESCRIPTION
**Pull request by a-beloglazov**

### CODE REVIEW

File: src/test/java/io/aiven/kafka/connect/gcs/testutils/BucketAccessor.java [hunk 1]
```diff
@@ -163,35 +170,42 @@ private String readStringContent0(final String blobName,
...
-    private InputStream getDecompressedStream(final InputStream inputStream,
-                                              final boolean compressed) throws IOException {
+    private InputStream getDecompressedStream(final InputStream inputStream, final String compression)
+            throws IOException {
         Objects.requireNonNull(inputStream, "inputStream cannot be null");
+        Objects.requireNonNull(compression, "compression cannot be null");
 
-        if (compressed) {
+        final CompressionType compressionType = CompressionType.forName(compression);
+        if (compressionType == CompressionType.ZSTD) {
```
Comment (ivanyu) [inline src/test/java/io/aiven/kafka/connect/gcs/testutils/BucketAccessor.java:179]: Could be `switch` here

File: src/test/java/io/aiven/kafka/connect/gcs/GcsSinkTaskTest.java [hunk 2]
```diff
@@ -395,14 +431,34 @@ private SinkRecord createNullRecord(final String topic,
...
+        blobNameValuesMap.put("topic0-2-50" + extension, toCollectionOfLists("value4", "value9"));
+        return blobNameValuesMap;
+    }
+
+    private Collection<List<String>> toCollectionOfLists(final String... values) {
+        return toCollectionOfLists(Lists.newArrayList(values));
+    }
+
+    private Collection<List<String>> toCollectionOfLists(final List<String> values) {
+        return values.stream().map(Collections::singletonList).collect(Collectors.toList());
```
Comment (ivanyu) [inline src/test/java/io/aiven/kafka/connect/gcs/GcsSinkTaskTest.java:462]: A short documentation with an example would be nice, would make it much easier to grasp quickly.

File: src/main/java/io/aiven/kafka/connect/gcs/GcsSinkTask.java [hunk 3]
```diff
@@ -133,6 +133,10 @@ private OutputStream getCompressedStream(final OutputStream outputStream) throws
 
         if (config.getCompressionType() == CompressionType.GZIP) {
             return new GZIPOutputStream(outputStream);
+        } else if (config.getCompressionType() == CompressionType.SNAPPY) {
+            return new SnappyOutputStream(outputStream);
+        } else if (config.getCompressionType() == CompressionType.ZSTD) {
+            return new ZstdOutputStream(outputStream);
         } else {
             return outputStream;
         }
```
Comment (ivanyu) [inline src/main/java/io/aiven/kafka/connect/gcs/GcsSinkTask.java:142]: Could be `switch`

File: src/integration-test/java/io/aiven/kafka/connect/gcs/IntegrationTest.java [hunk 4]
```diff
@@ -162,11 +159,13 @@ final void tearDown() {
         connectRunner.awaitStop();
     }
 
-    @Test
-    final void basicTest() throws ExecutionException, InterruptedException {
+    @ParameterizedTest
+    @ValueSource(strings = {"none", "gzip", "snappy", "zstd"})
+    final void basicTest(final String compression) throws ExecutionException, InterruptedException {
+        final CompressionType compressionType = CompressionType.forName(compression);
```
Comment (ivanyu) [inline src/integration-test/java/io/aiven/kafka/connect/gcs/IntegrationTest.java:165]: Seems not needed

File: src/integration-test/java/io/aiven/kafka/connect/gcs/IntegrationTest.java [hunk 5]
```diff
@@ -211,19 +210,21 @@ final void basicTest() throws ExecutionException, InterruptedException {
...
             }
         }
     }
 
-    @Test
-    final void groupByTimestampVariable() throws ExecutionException, InterruptedException {
+    @ParameterizedTest
+    @ValueSource(strings = {"none", "gzip", "snappy", "zstd"})
+    final void groupByTimestampVariable(final String compression) throws ExecutionException, InterruptedException {
+        final CompressionType compressionType = CompressionType.forName(compression);
```
Comment (ivanyu) [inline src/integration-test/java/io/aiven/kafka/connect/gcs/IntegrationTest.java:224]: Same here: seems not needed

File: src/integration-test/java/io/aiven/kafka/connect/gcs/IntegrationTest.java [hunk 6]
```diff
@@ -283,16 +284,20 @@ private String getTimestampBlobName(final int partition, final int startOffset)
                 TEST_TOPIC_0,
                 partition,
                 startOffset,
-                time.format(DateTimeFormatter.ofPattern("YYYY")),
+                time.format(DateTimeFormatter.ofPattern("yyyy")),
```
Comment (ivanyu) [inline src/integration-test/java/io/aiven/kafka/connect/gcs/IntegrationTest.java:287]: Nice catch, thanks!

File: src/integration-test/java/io/aiven/kafka/connect/gcs/IntegrationTest.java (hunk 4 above)
Comment (a-beloglazov) [reply src/integration-test/java/io/aiven/kafka/connect/gcs/IntegrationTest.java:165]: I have left only "none" compression. Is it ok?

File: src/test/java/io/aiven/kafka/connect/gcs/GcsSinkTaskTest.java [hunk 7]
```diff
@@ -395,14 +431,42 @@ private SinkRecord createNullRecord(final String topic,
...
+        return blobNameValuesMap;
+    }
+
+    /*
+     * example
+     * <pre>
+     * Collection<List<String>> expected = toCollectionOfLists("value0", "value5");
+     * Collection<List<String>> actual = readLinesFromBlob(blob, compression);
+     * assertIterableEquals(expected, actual);
+     * </pre>
```
Comment (ivanyu) [inline src/test/java/io/aiven/kafka/connect/gcs/GcsSinkTaskTest.java:463]: Sorry, when I asked for example, I meant not a usage example, but an example of input-output pair :) Like 
```
Input: "value0", "value5"
Output: Collection[List["value0"], List["value5"]]
```
Could you please add?

Comment (a-beloglazov) [reply src/test/java/io/aiven/kafka/connect/gcs/GcsSinkTaskTest.java]: Done

**Pull request merged**